# Hypothesis Testing

Q1 .  A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [2]:
data=pd.read_csv('C:\\Users\\ujwala\\Downloads\\Cutlets.csv')
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [3]:
unitA=pd.Series(data.iloc[:,0])
unitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [4]:
unitB=pd.Series(data.iloc[:,1])
unitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [5]:
#2 sample 2 # ind -> independent samples
p_value=stats.ttest_ind(unitA,unitB)
p_value

Ttest_indResult(statistic=0.7228688704678061, pvalue=0.4722394724599501)

In [6]:
p_value[1]     # 2-tail probability 

0.4722394724599501

In [44]:
# compare p_value with α = 0.05 (At 5% significance level)

Assume Null hyposthesis as Ho: μ1 = μ2 (There is no difference in diameters of cutlets between two units).

Thus Alternate hypothesis as Ha: μ1 ≠ μ2 (There is significant difference in diameters of cutlets between two units) 2 Sample 2 Tail test applicable

Q2 .A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 


In [9]:
data=pd.read_csv('C:\\Users\\ujwala\\Desktop\\ujwala Ds assignment\\Ds Datasets\\LabTAT.csv')
data.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [10]:
# Anova f test statistics: stats.f_oneway(column-1,column-2,column-3,column-4)
p_value=stats.f_oneway(data.iloc[:,0],data.iloc[:,1],data.iloc[:,2],data.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [11]:
p_value[1]  # compare it with α = 0.05

2.1156708949992414e-57

So the p-value[1] as campare to α = 0.05 is more 

Analysis of varaince between more than 2 samples or columns Assume Null Hypothesis Ho as No Varaince: All samples TAT population means are same

Thus Alternate Hypothesis Ha as It has Variance: Atleast one sample TAT population mean is different.

Q3 .Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.


In [13]:
import pandas as pd
from scipy import stats as stats
import numpy as np

In [14]:
data= pd.read_csv('C:\\Users\\ujwala\\Desktop\\ujwala Ds assignment\\Ds Datasets\\BuyerRatio.csv')

In [15]:
data.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [18]:
data_table=data.iloc[:,1:6]
data_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [19]:
data_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]], dtype=int64)

In [21]:
val=stats.chi2_contingency(data_table)
val

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [22]:
type(val)

tuple

In [23]:
no_of_rows=len(data_table.iloc[0:2,0])
no_of_columns=len(data_table.iloc[0,0:4])
degree_of_f=(no_of_rows-1)*(no_of_columns-1)
print('Degree of Freedom=',degree_of_f)

('Degree of Freedom=', 3)


In [24]:
Expected_value=val[3]

In [25]:
Expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

In [26]:
from scipy.stats import chi2
chi_square=sum([(o-e)**2/e for o,e in zip(data_table.values,Expected_value)])
chi_square_statestic=chi_square[0]+chi_square[1]
chi_square_statestic

1.5152956451130446

In [27]:
critical_value=chi2.ppf(0.95,3)
critical_value

7.814727903251179

In [28]:
if chi_square_statestic >= critical_value:
	print('Dependent (reject H0)')
else:
	print('Independent (fail to reject H0)')

Independent (fail to reject H0)


In [29]:
pvalue=1-chi2.cdf(chi_square_statestic,3)
pvalue

0.6787446296467897

In [30]:
if pvalue <= 0.05:
	print('Dependent (reject H0)')
else:
	print('Independent (fail to reject H0)')

Independent (fail to reject H0)


In [31]:
no_of_columns

4

In [32]:
no_of_rows

2

In [33]:
data_table=pd.crosstab(data['East'],data['Observed Values'])
data_table

Observed Values  Females  Males
East                           
50                     0      1
435                    1      0

In [35]:
data_table.values

array([[0, 1],
       [1, 0]], dtype=int64)

p_value =0.6787 
Hence p_value > α = 0.05 So we can not reject Null Hypothesis

Q4.TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


In [36]:
from scipy.stats import chi2_contingency

In [37]:
data=pd.read_csv('C:\\Users\\ujwala\\Desktop\\ujwala Ds assignment\\Ds Datasets\\Costomer+OrderForm.csv')
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
5     Error Free  Error Free  Error Free  Error Free
6     Error Free   Defective  Error Free  Error Free
7     Error Free  Error Free  Error Free  Error Free
8     Error Free  Error Free  Error Free  Error Free
9     Error Free  Error Free  Error Free  Error Free
10     Defective  Error Free  Error Free  Error Free
11    Error Free  Error Free  Error Free  Error Free
12    Error Free  Error Free  Error Free   Defective
13    Error Free  Error Free  Error Free  Error Free
14    Error Free  Error Free  Error Free  Error Free
15    Error Free  Error Free   Defective  Error Free
16    Error Free  Error Free  Error Free  Error Free
17    Error Free  Error Free  Error Free  Error Free
18    Error Free  Error Free  Error Free  Error Free
19     Defective  Error Free  Error Free   Defective
20    Error Free  Error Free  Error Free  Error Free
21    Error Free  Error Free   Defective  Error Free
22    Error Free  Error Free  Error Free  Error Free
23     Defective  Error Free  Error Free  Error Free
24    Error Free  Error Free  Error Free  Error Free
25    Error Free  Error Free  Error Free  Error Free
26    Error Free  Error Free  Error Free  Error Free
27    Error Free  Error Free  Error Free  Error Free
28    Error Free  Error Free  Error Free  Error Free
29    Error Free  Error Free  Error Free  Error Free
..           ...         ...         ...         ...
270   Error Free  Error Free  Error Free  Error Free
271   Error Free  Error Free  Error Free  Error Free
272   Error Free  Error Free  Error Free  Error Free
273   Error Free  Error Free  Error Free  Error Free
274   Error Free   Defective   Defective  Error Free
275   Error Free  Error Free  Error Free  Error Free
276   Error Free  Error Free  Error Free  Error Free
277   Error Free  Error Free  Error Free  Error Free
278   Error Free  Error Free  Error Free  Error Free
279   Error Free  Error Free  Error Free  Error Free
280   Error Free  Error Free  Error Free  Error Free
281   Error Free  Error Free  Error Free  Error Free
282   Error Free  Error Free  Error Free  Error Free
283   Error Free   Defective  Error Free  Error Free
284   Error Free  Error Free  Error Free  Error Free
285   Error Free  Error Free  Error Free   Defective
286   Error Free  Error Free  Error Free  Error Free
287   Error Free  Error Free  Error Free  Error Free
288   Error Free  Error Free   Defective  Error Free
289    Defective  Error Free  Error Free  Error Free
290   Error Free  Error Free  Error Free  Error Free
291    Defective  Error Free  Error Free  Error Free
292   Error Free   Defective  Error Free  Error Free
293   Error Free  Error Free  Error Free  Error Free
294   Error Free   Defective  Error Free  Error Free
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [38]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [39]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [40]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [41]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [42]:
# Make a contingency table
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [43]:
# Chi2 contengency independence test
chi2_contingency(obs) # o/p is (Chi2 stats value, p_value, data, expected obsvations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

Inference: As (p_value = 0.2771) > (α = 0.05); Accept Null Hypthesis i.e. Independence of categorical variables.Thus, customer order forms defective % does not varies by centre.
